__tests__/app.test.js

```diff
@@ -137,9 +137,93 @@ describe('GET /api/articles/:article_id', () => {
 })
 
 
+describe("Challenge 7 - POST /api/articles/:article_id/comments", () => {
+
+    test("status 201 - should respond with the posted comment when the correct properties are provided", () => {
+
+        const comment = {
+            username: "butter_bridge",
+            body: "Ey up lad fancy a jar?"
+        }
+
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send(comment)
+        .expect(201)
+        .then(({body}) => {
+        
+        expect(body.comment).toHaveProperty('comment_id', 19)
+        expect(body.comment).toHaveProperty('body', 'Ey up lad fancy a jar?')
+        expect(body.comment).toHaveProperty('article_id', 3)
+        expect(body.comment).toHaveProperty('author', 'butter_bridge' )
+        expect(body.comment).toHaveProperty('votes', 0)
+        expect(body.comment).toHaveProperty('created_at', expect.any(String))
+
+        }) 
+    })
+
+
+    test("sends a 400 when an invalid type request is sent" , () => {
 
+        const comment = {
+            username: "butter_bridge",
+            body: "Ey up lad fancy a jar?"
+        }
 
+        return request(app)
+        .post('/api/articles/notAnId/comments')
+        .send(comment)
+        .expect(400)
+        .then(({body}) => {
 
+        expect(body).toMatchObject({
+            msg:'400 - invalid type request'
+        })
+
+        })
+
+
+    })
 
+    test("sends a 400 an incomplete comment is sent" , () => {
 
+        const comment = {
+            
+            username: "butter_bridge",
+        }
+
+        return request(app)
+        .post('/api/articles/9000/comments')
+        .send(comment)
+        .expect(400)
+        .then(({body}) => {
+
+        expect(body).toMatchObject({
+            msg:'400 - not found'
+        })
+        })
+
+    })
+    
+    test("sends a 404 when an invalid article_id number is sent" , () => {
+
+        const comment = {
+            username: "butter_bridge",
+            body: "Ey up lad fancy a jar?"
+        }
+
+        return request(app)
+        .post('/api/articles/9000/comments')
+        .send(comment)
+        .expect(404)
+        .then(({body}) => {
+
+        expect(body).toMatchObject({
+            msg:'404 - not found'
+        })
+
+        })
```

Comment:
There's a few other instances I'd like you to think of: 
201 - ignores unnecessary properties on the comment (i.e. it has a username, a body and a colour property)
404 - username does not exist

Reply:
Added :)